src/librustc_lexer/src/lib.rs

```diff
@@ -235,16 +235,15 @@ pub enum Base {
 /// Line won't be skipped if it represents a valid Rust syntax
 /// (e.g. "#![deny(missing_docs)]").
 pub fn strip_shebang(input: &str) -> Option<usize> {
-    debug_assert!(!input.is_empty());
-    let s: &str = &remove_whitespace(input);
-    if !s.starts_with("#!") || s.starts_with("#![") {
-        return None;
+    if let Some(first_line) = input.lines().next() {
+        let prefix = first_line.chars().filter(|c| !is_whitespace(*c)).collect::<Vec<_>>();
+        if !prefix.starts_with(&['#', '!']) || prefix.starts_with(&['#', '!', '[']) {
```

Comment:
Wait, this seems to allow a space between `#` and `!` or even *before* `#`, but AFAIK the only place where space is allowed is *after* the `!`.

So all of this seems overcomplicated. Maybe we should revert to the original version, before the last PR, and make the smallest change necessary?

Reply:
Maybe I misunderstand, but `# ! [attrib(...) ]` is valid rust? Do you mean that shebangs can't have a space there?

Reply:
I mean that shebangs can't have a space  there.